Assets/Engine/Inventory/Editor/ContainerDescriptorEditor.cs

```diff
@@ -341,15 +352,32 @@ private void AddAttached()
 
     private void AddSync()
     {
-        if(containerDescriptor.gameObject.GetComponent<ContainerSync>() == null)
+        // put the containersync on the highest game object in the hierarchy with a network identity
+        if (containerDescriptor.gameObject.GetComponentInParent<ContainerSync>() == null)
         {
             SerializedProperty sp = serializedObject.FindProperty("containerSync");
-            sp.objectReferenceValue = containerDescriptor.gameObject.AddComponent<ContainerSync>();
-            serializedObject.ApplyModifiedProperties();           
+            GameObject g = GetParentNetworkIdentity(containerDescriptor.gameObject);
+
+            if (g != null)
+            {
+                sp.objectReferenceValue = g.AddComponent<ContainerSync>();
+            }
+            else
+            {
+                sp.objectReferenceValue = containerDescriptor.gameObject.AddComponent<ContainerSync>();
+            }
```

Comment:
nitpick: avoid cryptic single letter variables as per code style guidelines. Also inline the if/else.

Reply:
done